src/main/scala/lms/transformation/distributed_tensor/distributedTensorConv.scala

```diff
@@ -102,14 +102,16 @@ trait FixedSizeDistributedTensorConvTypeLess extends FixedSizeDistributedTensorM
         // save forward op in forwardNodes
         forwardNodes += node
         // save backward op in backwardNodes
-        val x = new TENSOR(transform(a))      // weight
```

Comment:
the `new TENSOR` call has to be in the closure because they must be called later, not now